### PR TITLE
[receiver/skywalking]: Refactor code structure/directory

### PR DESCRIPTION
**Pull request by aheling11**
**Description:** <Describe what has changed.>

1st PR: In order to add metrics receiver to skywalking/receiver, I found that I need to adjust the code directory/code structure a bit. So this PR is just a code refactoring, referring to the structure of otlp/receiver. After that, it will be easier for skywalking/receiver to add metrics or logs to the receiver.

The new swReceiver struct is shown below.
``` 
type swReceiver struct {
	config *configuration

	grpc            *grpc.Server
	collectorServer *http.Server

	goroutines sync.WaitGroup

	settings receiver.CreateSettings

	traceReceiver *trace.Receiver
        metricsReceiver *metrics.Receiver // will be implemented in following PR
        

	dummyReportService *dummyReportService
}
```

**Link to tracking Issue:** <Issue number if applicable>
#20315 

**Testing:** <Describe what testing was performed and which tests were added.>
The original unit tests have been passed, and a test for HTTP reception has been added. Because of some restructuring, the unit test code has changed a bit.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by aheling11** [comment]
Sorry for adding a lot of other tags by accident, can someone help me to remove these labels?

**Comment by atoulme** [comment]
Fixed, no worries

**Comment by jpkrohling** [comment]
Assigned this to the code owner, @JaredTan95.

File: receiver/skywalkingreceiver/internal/trace/tracing_report_service.go
Comment (JaredTan95) [inline receiver/skywalkingreceiver/internal/trace/tracing_report_service.go:48]: how about rename `New` as `NewReceiver`?

**Review by JaredTan95 — CHANGES_REQUESTED** [review]

**Comment by JaredTan95** [comment]
@aheling11 thx for your contribution, and waiting for your next `mextircs receiver` impl. I added some nits, others LGTM :-P

File: receiver/skywalkingreceiver/internal/trace/tracing_report_service.go
Comment (aheling11) [reply receiver/skywalkingreceiver/internal/trace/tracing_report_service.go]: make sense. I'll rename it

**Review by JaredTan95 — APPROVED** [review]
LGTM

**Comment by JaredTan95** [comment]
@aheling11 CI passed, Let's wait for the approver to merge this PR.

**Comment by aheling11** [comment]
There were some conflicts in the go.sum, but they have been resolved.

**Comment by aheling11** [comment]
Could you please rerun the tests? I have resolved some conflicts.  @jpkrohling

**Comment by JaredTan95** [comment]
@aheling11 new conflicts :-P

**Comment by aheling11** [comment]
> @aheling11 new conflicts :-P

resolved again ~

**Review by JaredTan95 — APPROVED** [review]

**Comment by JaredTan95** [comment]
That's ok, take easy~, let's waiting maintainer to merge it.

**Review by fatsheep9146 — APPROVED** [review]

**Comment by aheling11** [comment]
resolve conflict again.

**Comment by fatsheep9146** [comment]
ping @open-telemetry/collector-contrib-maintainer  I think this pr is ready to be merged

**Review by atoulme — APPROVED** [review]

**Pull request merged**
